table/tables/partition_test.go

```diff
@@ -700,6 +702,1460 @@ func TestIssue31629(t *testing.T) {
 	}
 }
 
+type compoundSQL struct {
+	selectSQL        string
+	point            bool
+	batchPoint       bool
+	pruned           bool
+	executeExplain   bool
+	usedPartition    []string
+	notUsedPartition []string
+	rowCount         int
+}
+
+type partTableCase struct {
+	partitionbySQL string
+	selectInfo     []compoundSQL
+}
+
+func executePartTableCase(t *testing.T, tk *testkit.TestKit, testCases []partTableCase,
+	createSQL string, insertSQLs []string, dropSQL string) {
+	for i, testCase := range testCases {
+		// create table ... partition by key ...
+		ddlSQL := createSQL + testCase.partitionbySQL
+		fmt.Println(i, ":", ddlSQL)
+		executeSQLWrapper(t, tk, ddlSQL)
+		// insert data
+		for _, insertsql := range insertSQLs {
+			executeSQLWrapper(t, tk, insertsql)
+		}
+		// execute testcases
+		for j, selInfo := range testCase.selectInfo {
+			fmt.Println(j, ":", selInfo.selectSQL)
+			tk.MustQuery(selInfo.selectSQL).Check(testkit.Rows(strconv.Itoa(selInfo.rowCount)))
+			if selInfo.executeExplain {
+				result := tk.MustQuery("EXPLAIN " + selInfo.selectSQL)
+				if selInfo.point {
+					result.CheckContains("Point_Get")
```

Comment:
Maybe use `HasPlan` or `HasPlan4ExplainFor` instead?

Reply:
Yes, it's more accurate, but we need to check so many different scenes, such as `execute plan`, `error type` ...., 
maybe `CheckContains` is more `flexible`.